language/move-prover/docgen/src/docgen.rs

```diff
@@ -0,0 +1,930 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#[allow(unused_imports)]
+use log::{info, warn};
+
+use itertools::Itertools;
+use num::BigUint;
+use regex::Regex;
+use serde::Serialize;
+use spec_lang::{
+    ast::{ModuleName, SpecBlockInfo, SpecBlockTarget},
+    code_writer::CodeWriter,
+    emit, emitln,
+    env::{FunctionEnv, GlobalEnv, ModuleEnv, Parameter, StructEnv, TypeConstraint, TypeParameter},
+    symbol::Symbol,
+    ty::TypeDisplayContext,
+};
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, VecDeque},
+    rc::Rc,
+};
+
+const KEYWORDS: &[&str] = &[
+    "abort",
+    "acquires",
+    "as",
+    "break",
+    "continue",
+    "copy",
+    "copyable",
+    "",
+    "else",
+    "false",
+    "fun",
+    "if",
+    "invariant",
+    "let",
+    "loop",
+    "module",
+    "move",
+    "native",
+    "public",
+    "resource",
+    "return",
+    "spec",
+    "struct",
+    "true",
+    "use",
+    "while",
+];
+
+const WEAK_KEYWORDS: &[&str] = &[
+    "apply",
+    "assert",
+    "assume",
+    "decreases",
+    "aborts_if",
+    "ensures",
+    "requires",
+    "pack",
+    "unpack",
+    "update",
+    "to",
+    "schema",
+    "include",
+    "old",
+    "except",
+    "global",
+    "mut",
```

Comment:
- I was surprised to learn that this is not a keyword!
- @tnowacki: do global state accessors like `exists`, `borrow_global`, etc belong in weak keywords?

Reply:
I was just taking this from the fmt function on lexer.rs Tok. Not sure whether it is complete.